src/extension/background-script/actions/webln/getBalanceOrPrompt.ts

```diff
@@ -0,0 +1,83 @@
+import utils from "~/common/lib/utils";
+import db from "~/extension/background-script/db";
+import state from "~/extension/background-script/state";
+import { MessageDefault } from "~/types";
+
+const getBalanceOrPrompt = async (message: MessageDefault) => {
```

Comment:
This prompt / allowance / permission logic would be great to be extracted - most of the lines are the same across different actions